sql/testdata/distinct

```diff
@@ -0,0 +1,55 @@
+statement ok
+CREATE TABLE xyz (
+  x INT PRIMARY KEY,
+  y INT,
+  z INT,
+  INDEX foo (z, y)
+)
+
+statement ok
+INSERT INTO xyz VALUES (1, 2, 3), (2, 5, 6), (3, 2, 3), (4, 5, 6), (5, 2, 6), (6, 3, 5), (7, 2, 9);
+
+query II
+SELECT y, z FROM xyz
+----
+2 3
+5 6
+2 3
+5 6
+2 6
+3 5
+2 9
+
+query II
+SELECT DISTINCT y, z FROM xyz
+----
+2 3
+5 6
+2 6
+3 5
+2 9
+
+query II
+SELECT DISTINCT y, z FROM xyz ORDER BY z
+----
+2 3
+3 5
+2 6
+5 6
+2 9
+
+query II
+SELECT DISTINCT y, z FROM xyz ORDER BY y, z
+----
+2 3
+2 6
+2 9
+3 5
+5 6
+
+query II
+SELECT DISTINCT y + z FROM xyz ORDER by (y + z)
+----
+5
+8
+11
```

Comment:
We should also have a test of the interaction between DISTINCT and NULL. `NULL == NULL` is false, but `encodeKey(DNull) == encodeKey(DNull)`. Are rows containing NULLs collapsed by DISTINCT or not? (it looks like they are in other DBs, so the implementation is doing the right thing, but we should have a test).